pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -190,6 +234,29 @@ func (i *Controller) handleUpdatePod(old interface{}, newObj interface{}) {
 		utilruntime.HandleError(err)
 		return
 	}
+
+	oldPodIp := old.(*k8sv1.Pod).Status.PodIP
+	updatedPodIp := newObj.(*k8sv1.Pod).Status.PodIP
+	podHostIP := newObj.(*k8sv1.Pod).Status.HostIP
+	// Pod events that are skipped during addEvent are handled here when they are assigned an ipaddress.
+	if updatedPodIp != "" && oldPodIp != updatedPodIp {
+		klog.V(4).Infof("In handleUpdatePod, pod %s is now assigned %s address, enqueing", old.(*k8sv1.Pod).Name, updatedPodIp)
+		i.enqueueObject(newObj, i.podWorkqueue)
```

Comment:
you can just use the key which we already have and add it directly without needing to call enqueueObject again. will also avoid the log msg in enqueueObject.

Reply:
No, I meant add it to queue. Yeah, just realized that we can simplify it. After refactoring I did, we don't really need to call it. Maybe can add enqueueKey method for such scenarios.